biz.aQute.bndlib/src/aQute/bnd/osgi/FileResource.java

```diff
@@ -51,7 +52,7 @@ public String toString() {
 	}
 
 	public void write(OutputStream out) throws Exception {
-		copy(this, out);
+		Files.copy(file.toPath(), out);
```

Comment:
I am not opposed the change per se. My main concern was leaving dead code behind (I hate it when people do that). So make the change, but move the copy method logic into ZipResource.write.